src/ruby/PreProcessor.rb

```diff
@@ -47,7 +47,7 @@ def process_emerald(input)
   def check_new_indent(new_indent)
     if new_indent > @current_indent
       new_indent_greater(new_indent)
-    elsif new_indent < @current_indent && new_indent.nonzero?
```

Comment:
Ahh yeah, this isn't necessary because I added `next if line.lstrip.empty?`. Thanks for catching that!